### PR TITLE
ci: Fixing Spanner CDC correctness tests and correcting the Failure injection tests workflow

### PR DESCRIPTION
**Pull request by darshan-sj**
Successful runs:
1. https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/runs/25712969377/job/75496883229
2. https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/runs/25712980265/job/75496919206
3. https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/runs/25712990317/job/75496949559

### CODE REVIEW

File: v2/failure-injection-policies/src/main/java/com/google/cloud/teleport/v2/failureinjection/TransactionTimeoutInjectionPolicy.java [hunk 1]
```diff
@@ -85,6 +87,26 @@ public TransactionTimeoutInjectionPolicy(JsonNode inputParameter) {
...
+          LOG.info("Job start time provided: {}", this.startTime);
+        } catch (DateTimeParseException e) {
+          throw new IllegalArgumentException(
+              "Failed to parse "
+                  + JOB_START_TIME_FIELD
+                  + ". Expected ISO-8601 format. Value: "
+                  + inputParameter.get(JOB_START_TIME_FIELD).asText(),
+              e);
+        }
+      }
```
[the remark this answers is not in the document]
Comment (darshan-sj) [reply v2/failure-injection-policies/src/main/java/com/google/cloud/teleport/v2/failureinjection/TransactionTimeoutInjectionPolicy.java:109]: This is a very minor optimization for a corner case.

File: v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/failureinjectiontesting/DataStreamToSpannerCDCFT.java [hunk 2]
```diff
@@ -192,12 +192,15 @@ public void setUp() throws IOException, InterruptedException {
...
+    String jobStartTime = java.time.Instant.now().toString();
     FlexTemplateDataflowJobResourceManager.Builder flexTemplateBuilder =
         FlexTemplateDataflowJobResourceManager.builder(testName)
             .withAdditionalMavenProfile("failureInjectionTest")
             .addParameter(
                 "failureInjectionParameter",
-                "{\"policyType\":\"TransactionTimeoutInjectionPolicy\", \"policyInput\": { \"injectionWindowDuration\": \"PT30M\", \"delayDuration\": \"PT260S\" }}")
+                "{\"policyType\":\"TransactionTimeoutInjectionPolicy\", \"policyInput\": { \"injectionWindowDuration\": \"PT30M\", \"delayDuration\": \"PT260S\", \"jobStartTime\": \""
+                    + jobStartTime
+                    + "\" }}")
```
[the remark this answers is not in the document]
Comment (darshan-sj) [reply v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/failureinjectiontesting/DataStreamToSpannerCDCFT.java]: Done.

File: v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/failureinjectiontesting/SpannerToSrcDBMySQLCDCFT.java [hunk 3]
```diff
@@ -164,7 +165,9 @@ public void setUp() throws IOException, InterruptedException {
                 "maxShardConnections",
                 MAX_WORKERS,
                 "failureInjectionParameter",
-                "{\"policyType\":\"TransactionTimeoutInjectionPolicy\", \"policyInput\": { \"injectionWindowDuration\": \"PT30M\", \"delayDuration\": \"PT80S\" }}"),
+                "{\"policyType\":\"TransactionTimeoutInjectionPolicy\", \"policyInput\": { \"injectionWindowDuration\": \"PT30M\", \"delayDuration\": \"PT80S\", \"jobStartTime\": \""
+                    + jobStartTime
+                    + "\" }}"),
```
[the remark this answers is not in the document]
Comment (darshan-sj) [reply v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/failureinjectiontesting/SpannerToSrcDBMySQLCDCFT.java]: Done.

**Review by aasthabharill — APPROVED** [review]
